dynamic_programming/all_construct.py

```diff
@@ -0,0 +1,56 @@
+"""
+Program to list all the ways a target string can be
+constructed from the given list of substrings
+"""
+
+
+def all_construct(target: str, word_bank: list[str] = None) -> list[list[str]]:
+    """
+        returns the list containing all the possible
+        combinations a string(target) can be constructed from
+        the given list of substrings(word_bank)
+    >>> all_construct("hello", ["he", "l", "o"])
+    [['he', 'l', 'l', 'o']]
+    >>> all_construct("purple",["purp","p","ur","le","purpl"])
+    [['purp', 'le'], ['p', 'ur', 'p', 'le']]
+    """
+
+    if word_bank is None:
+        word_bank = []
```

Comment:
Safer in the presence of other falsey values...
```suggestion
    word_bank  = word_bank or []
```

Reply:
Ok this is cleaner. Will update.